File: backend/src/entities/Student.ts
import { Entity, PrimaryGeneratedColumn, Column, OneToMany, Unique } from 'typeorm';
import { Attendance } from './Attendance';
import { PermanentAbsence } from './PermanentAbsence';
import { StudentMonthlyOverride } from './StudentMonthlyOverride';
import { Claim } from './Claim';

@Entity('students')
@Unique(['student_number'])
@Unique(['nickname'])
export class Student {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ unique: true })
  student_number: string;

  @Column({ unique: true })
  nickname: string;

  @Column()
  first_name: string;

  @Column()
  last_name: string;

  @Column({ nullable: true })
  phone_number: string;

  @Column({ type: 'enum', enum: ['יסודי', 'תיכון'] })
  school_level: 'יסודי' | 'תיכון';

  @Column({ type: 'enum', enum: ['פעיל', 'לא פעיל', 'מושעה'], default: 'פעיל' })
  activity_status: 'פעיל' | 'לא פעיל' | 'מושעה';

  @OneToMany(() => Attendance, attendance => attendance.student)
  attendances: Attendance[];

  @OneToMany(() => PermanentAbsence, absence => absence.student)
  permanent_absences: PermanentAbsence[];

  @OneToMany(() => StudentMonthlyOverride, override => override.student)
  monthly_overrides: StudentMonthlyOverride[];

  @OneToMany(() => Claim, claim => claim.student)
  claims: Claim[];
}

File: backend/app/models/student.py
from sqlalchemy import Column, Integer, String, Enum
from sqlalchemy.orm import relationship
from .base import Base
import enum

class SchoolLevel(enum.Enum):
    ELEMENTARY = "יסודי"
    HIGH_SCHOOL = "תיכון"

class ActivityStatus(enum.Enum):
    ACTIVE = "פעיל"
    INACTIVE = "לא פעיל"
    SUSPENDED = "מושעה"

class Student(Base):
    __tablename__ = "students"
    
    id = Column(Integer, primary_key=True, index=True)
    student_number = Column(String, unique=True, nullable=False, index=True)
    nickname = Column(String, unique=True, nullable=True, index=True)
    first_name = Column(String, nullable=False)
    last_name = Column(String, nullable=False)
    phone_number = Column(String, nullable=True)
    school_level = Column(Enum(SchoolLevel), nullable=False)
    activity_status = Column(Enum(ActivityStatus), nullable=False, default=ActivityStatus.ACTIVE)
    
    # Relationships
    attendances = relationship("Attendance", back_populates="student")
    permanent_absences = relationship("PermanentAbsence", back_populates="student")
    monthly_overrides = relationship("StudentMonthlyOverride", back_populates="student")
    claims = relationship("Claim", back_populates="student")